bioconda_utils/lint/__init__.py

```diff
@@ -0,0 +1,446 @@
+"""Recipe Linter
+
+Writing additional checks
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Lint checks are defined in :py:mod:`bioconda_utils.lint.checks` as
+subclasses of `LintCheck`. It might be easiest to have a look at that
+module and the already existing checks and go from there.
+
+Briefly, each class becomes a check by:
+
+- The class name becomes the name of the check in the documentation
+  and when skipping lints. The convention is to use lower case
+  separated by underscore.
+
+- The docstring is used to describe the check on the command line,
+  on Github when the check fails and in the documentation.
+
+  The first line is considered a title or one line summary used where
+  a brief description is needed. The remainder is a long desription
+  and should include brief help on how to fix the detected issue.
+
+- The class property ``severity`` defaults to ``ERROR`` but can be
+  set to ``INFO`` or ``WARNING`` for informative checks that
+  should not cause linting to fail.
+
+- The class property ``requires`` may contain a list of other check
+  classes that are required to have passed before this check is
+  executed. Use this to avoid duplicate errors presented, or to
+  ensure that asumptions made by your check are met by the recipe.
+
+- Each class is instantiated once per linting run. Do slow preparation
+  work in the constructor. E.g. the `recipe_is_blacklisted` check
+  loads and parses the blacklist here.
+
+- As each recipe is linted, your check will get called on three
+  functions: `check_recipe <LintCheck.check_recipe>`, `check_deps
+  <LintCheck.check_deps>` and `check_source <LintCheck.check_source>`.
+
+  The first is simply passed the `Recipe` object, which you can use to
+  inspect the recipe in question. The latter two are for convenience
+  and performance. The ``check_deps`` call receives a mapping of
+  all dependencies mentioned in the recipe to a list of locations
+  in which the dependency occurred. E.g.::
+
+    {'setuptools': ['requirements/run',
+                    'outputs/0/requirements/run']}
+
+  The ``check_sources`` is called for each source listed in the
+  recipe, whether ``source:`` is a dict or a source, eliminating the
+  need to repeat handling of these two cases in each check. The value
+  ``source`` passed is a dictionary of the source, the value
+  ``section`` a path to the source section.
+
+- When a check encounters an issue, it should use ``self.message()`` to
+  record the issue. The message is commonly modified with a path in
+  the meta data to point to the part of the ``meta.yaml`` that lead to
+  the error. This will be used to position annotations on github.
```

Comment:
...and if no message is created, the check is assumed to have passed.

Reply:
done

Reply:
Fixed in dc5c91b